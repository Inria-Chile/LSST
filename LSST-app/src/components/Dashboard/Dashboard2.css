#dashboard {
    height: 100vh;
    width: 100vw;
    overflow: hidden;
}

#dashboard-content {
    display: inline-block;
}

#dashboard-toolbar{
    height: 100%;
    background: #2d4053;
    display: inline-block;    
}

#dashboard-content > div {
    background: #2d4053;
    margin: 2px;
    cursor: default;
}

#toolbar {
    width: 3vw;
}

.field-details-popup {
    box-shadow: 5px 5px 20px rgba(0,0,0,0.5);    
}

.react-draggable {
    overflow: hidden;
    box-shadow: 0px 0px 15px rgba(0,0,0,0.3);
}

.react-draggable-dragging {
    overflow: hidden;
    box-shadow: 5px 5px 30px black;
    transition: box-shadow 0.3s ease-in-out;
    z-index: 100;
}

.field-details-popup.react-draggable-dragging {
    overflow: hidden;
    box-shadow: 10px 10px 60px black;
    transition: box-shadow 0.3s ease-in-out;
    z-index: 100;
}