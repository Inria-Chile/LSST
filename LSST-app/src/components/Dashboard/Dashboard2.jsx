import React, { Component } from 'react';
// import ReactDOM from 'react-dom';
import Survey from '../Survey/Survey';
import Mirrors from '../Mirrors/Mirrors';
import Dome from '../Dome/Dome';
import Environment from '../Environment/Environment';
import FieldDetails from '../Survey/FieldDetails/FieldDetails';
import {Responsive, WidthProvider} from 'react-grid-layout';
import './Dashboard2.css';
const ReactGridLayout = WidthProvider(Responsive);

class Dashboard2 extends Component {
    
    static viewComponents = [Survey, Mirrors];
    static layout = {
        'survey': {i: 'survey', x: 0, y: 0, w: 6, h: 10},
        'mirrors': {i: 'mirrors', x: 6, y: 0, w: 2, h: 12},
        'dome': {i: 'dome', x: 8, y: 0, w: 4, h: 20},
        'environment': {i: 'environment', x: 0, y: 10, w: 3, h: 3}
    };
    constructor(props){
        super(props);
        this.state = {
            showFieldDetails: false,
            environmentOpenRows: 0,
            layout: Object.values(Dashboard2.layout),
        }
    }

    setFieldDetailsVisibility = (visibility) => {
        this.setState({
            showFieldDetails: visibility,
        });
    }

    setSelectedFieldData = (data) => {
        this.setState({
            selectedFieldData: data
        });
    }

    setComponentHeight = (component, height) => {
        let newLayout = JSON.parse(JSON.stringify(this.state.layout));
        newLayout.map((l, i) => {
            if(l.i === component)
                newLayout[i] = {...newLayout[i], h:height};
            return l;
        });
        this.setState({
            layout: newLayout
        });
    }

    setEnvironmentOpenRows = (rows) => {
        this.setComponentHeight('environment', 3+rows);
    }

    onLayoutChange = (layout) => {
        this.setState({layout: layout});
    }

    render() {
        return (
            <div id="dashboard" ref={(dashboard) => {
                if(!this.state.dashboardNode){
                    this.setState({
                        dashboardNode: dashboard
                    })
                }
            }}>
                <div id="dashboard-content2">
                    <ReactGridLayout className="layout" layouts={{lg: this.state.layout}} rowHeight={90} width={3840}
                        breakpoints={{lg: 3200, md: 996, sm: 768, xs: 480, xxs: 0}} draggableHandle={'.move-button'}
                        cols={{lg: 12, md: 6, sm: 6, xs: 6, xxs: 6}} onLayoutChange={this.onLayoutChange}>

                        <div key="survey">
                            <Survey parentNode={this.state.dashboardNode} 
                                    setFieldDetailsVisibility={this.setFieldDetailsVisibility} 
                                    showFieldDetails={this.state.showFieldDetails}
                                    setSelectedFieldData={this.setSelectedFieldData}/>
                        </div>
                        <div key="mirrors">
                            <Mirrors />
                        </div>
                        <div key="dome">
                            <Dome />
                        </div>
                        <div key="environment">
                            <Environment setOpenRows={this.setEnvironmentOpenRows}/>
                        </div>
                    </ReactGridLayout>
                </div>
                    {
                        this.state.showFieldDetails ? 
                        <FieldDetails fieldData={this.state.selectedFieldData}
                                        setFieldDetailsVisibility={this.setFieldDetailsVisibility}
                                        showFieldDetails={this.state.showFieldDetails}/>
                        :
                        ''
                    }
            </div>
        );
    }
}

export default Dashboard2;
