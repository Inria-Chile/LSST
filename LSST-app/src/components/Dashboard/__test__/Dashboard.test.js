import React from 'react';
import Dashboard2 from '../Dashboard2';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from "enzyme"; 
import toJson from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });


describe('Dahshboard test',function(){
    it('renders correctly with shallow renderer',()=>{
        const tree = shallow(<Dashboard2/>);
        expect(toJson(tree)).toMatchSnapshot();
    });

    it('renders FieldDetails when showFieldDetails is setted true',()=>{
        let tree  = shallow(<Dashboard2/>);
        tree.setState({showFieldDetails : true});
        expect(toJson(tree)).toMatchSnapshot();
    });

    it('doesnt render FieldDetails when showFieldDetails is setted false',()=>{
        let tree  = shallow(<Dashboard2/>);
        tree.setState({showFieldDetails : false});
        expect(toJson(tree)).toMatchSnapshot();
    });

});