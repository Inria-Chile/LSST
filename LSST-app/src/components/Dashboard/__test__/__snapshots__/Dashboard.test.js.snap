// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dahshboard test doesnt render FieldDetails when showFieldDetails is setted false 1`] = `
<div
  id="dashboard"
>
  <div
    id="dashboard-content2"
  >
    <_class
      breakpoints={
        Object {
          "lg": 3200,
          "md": 996,
          "sm": 768,
          "xs": 480,
          "xxs": 0,
        }
      }
      className="layout"
      cols={
        Object {
          "lg": 12,
          "md": 6,
          "sm": 6,
          "xs": 6,
          "xxs": 6,
        }
      }
      draggableHandle=".move-button"
      layout={
        Array [
          Object {
            "h": 5,
            "i": "a",
            "w": 6,
            "x": 0,
            "y": 0,
          },
          Object {
            "h": 2,
            "i": "b",
            "maxW": 4,
            "minW": 2,
            "w": 3,
            "x": 1,
            "y": 0,
          },
          Object {
            "h": 2,
            "i": "c",
            "w": 1,
            "x": 4,
            "y": 0,
          },
        ]
      }
      measureBeforeMount={false}
      rowHeight={90}
      width={3840}
    >
      <div
        data-grid={
          Object {
            "h": 10,
            "w": 6,
            "x": 0,
            "y": 0,
          }
        }
        key="a"
      >
        <Survey
          setFieldDetailsVisibility={[Function]}
          setSelectedFieldData={[Function]}
          showFieldDetails={false}
        />
      </div>
      <div
        data-grid={
          Object {
            "h": 12,
            "w": 2,
            "x": 6,
            "y": 0,
          }
        }
        key="b"
      >
        <Mirrors />
      </div>
      <div
        data-grid={
          Object {
            "h": 19,
            "w": 4,
            "x": 8,
            "y": 0,
          }
        }
        key="c"
      >
        <Dome />
      </div>
      <div
        data-grid={
          Object {
            "h": 5,
            "w": 3,
            "x": 0,
            "y": 10,
          }
        }
        key="d"
      >
        <Environment />
      </div>
    </_class>
  </div>
</div>
`;

exports[`Dahshboard test renders FieldDetails when showFieldDetails is setted true 1`] = `
<div
  id="dashboard"
>
  <div
    id="dashboard-content2"
  >
    <_class
      breakpoints={
        Object {
          "lg": 3200,
          "md": 996,
          "sm": 768,
          "xs": 480,
          "xxs": 0,
        }
      }
      className="layout"
      cols={
        Object {
          "lg": 12,
          "md": 6,
          "sm": 6,
          "xs": 6,
          "xxs": 6,
        }
      }
      draggableHandle=".move-button"
      layout={
        Array [
          Object {
            "h": 5,
            "i": "a",
            "w": 6,
            "x": 0,
            "y": 0,
          },
          Object {
            "h": 2,
            "i": "b",
            "maxW": 4,
            "minW": 2,
            "w": 3,
            "x": 1,
            "y": 0,
          },
          Object {
            "h": 2,
            "i": "c",
            "w": 1,
            "x": 4,
            "y": 0,
          },
        ]
      }
      measureBeforeMount={false}
      rowHeight={90}
      width={3840}
    >
      <div
        data-grid={
          Object {
            "h": 10,
            "w": 6,
            "x": 0,
            "y": 0,
          }
        }
        key="a"
      >
        <Survey
          setFieldDetailsVisibility={[Function]}
          setSelectedFieldData={[Function]}
          showFieldDetails={true}
        />
      </div>
      <div
        data-grid={
          Object {
            "h": 12,
            "w": 2,
            "x": 6,
            "y": 0,
          }
        }
        key="b"
      >
        <Mirrors />
      </div>
      <div
        data-grid={
          Object {
            "h": 19,
            "w": 4,
            "x": 8,
            "y": 0,
          }
        }
        key="c"
      >
        <Dome />
      </div>
      <div
        data-grid={
          Object {
            "h": 5,
            "w": 3,
            "x": 0,
            "y": 10,
          }
        }
        key="d"
      >
        <Environment />
      </div>
    </_class>
  </div>
  <FieldDetails
    setFieldDetailsVisibility={[Function]}
    showFieldDetails={true}
  />
</div>
`;

exports[`Dahshboard test renders correctly with shallow renderer 1`] = `
<div
  id="dashboard"
>
  <div
    id="dashboard-content2"
  >
    <_class
      breakpoints={
        Object {
          "lg": 3200,
          "md": 996,
          "sm": 768,
          "xs": 480,
          "xxs": 0,
        }
      }
      className="layout"
      cols={
        Object {
          "lg": 12,
          "md": 6,
          "sm": 6,
          "xs": 6,
          "xxs": 6,
        }
      }
      draggableHandle=".move-button"
      layout={
        Array [
          Object {
            "h": 5,
            "i": "a",
            "w": 6,
            "x": 0,
            "y": 0,
          },
          Object {
            "h": 2,
            "i": "b",
            "maxW": 4,
            "minW": 2,
            "w": 3,
            "x": 1,
            "y": 0,
          },
          Object {
            "h": 2,
            "i": "c",
            "w": 1,
            "x": 4,
            "y": 0,
          },
        ]
      }
      measureBeforeMount={false}
      rowHeight={90}
      width={3840}
    >
      <div
        data-grid={
          Object {
            "h": 10,
            "w": 6,
            "x": 0,
            "y": 0,
          }
        }
        key="a"
      >
        <Survey
          setFieldDetailsVisibility={[Function]}
          setSelectedFieldData={[Function]}
          showFieldDetails={false}
        />
      </div>
      <div
        data-grid={
          Object {
            "h": 12,
            "w": 2,
            "x": 6,
            "y": 0,
          }
        }
        key="b"
      >
        <Mirrors />
      </div>
      <div
        data-grid={
          Object {
            "h": 19,
            "w": 4,
            "x": 8,
            "y": 0,
          }
        }
        key="c"
      >
        <Dome />
      </div>
      <div
        data-grid={
          Object {
            "h": 5,
            "w": 3,
            "x": 0,
            "y": 10,
          }
        }
        key="d"
      >
        <Environment />
      </div>
    </_class>
  </div>
</div>
`;
